src/software/backend/input/network/filter/ball_filter.cpp

```diff
@@ -323,8 +320,8 @@ std::optional<BallState> BallFilter::estimateBallState(
         // more robust to fluctuations in ball position, so will not vary as much as using
         // the "raw" ball velocity
         auto velocity_direction_along_regression_line =
-            velocity_estimate->average_velocity.project(regression_line.getSecond() -
-                                                        regression_line.getFirst());
+            velocity_estimate->average_velocity.project(
+                regression_line.toNormalUnitVector().perpendicular());
```

Comment:
`toNormalUnitVector` produces a vector that points in a direction perpendicular to the `Line`, so wouldn't `perpendicular` be necessary to make it point in the direction of regression line?

Reply:
Ah you're right in that case. I read is as it converts the line to a normalized unit vector, where "normalized" means "unit vector" as well (because we use `norm` to create unit vectors in the `Vector` class).

This should be fine then.